File: python/examples/with_playwright.py
import sys

from playwright.sync_api import sync_playwright

from captchakraken import PageSolver

URL = sys.argv[1] if len(sys.argv) > 1 else "https://www.google.com/recaptcha/api2/demo"


def main() -> int:
    with sync_playwright() as p:
        browser = p.chromium.launch(headless=True)
        page = browser.new_page()
        page.goto(URL)

        result = PageSolver().solve(page)

        print("✅ solved" if result.is_solved else "❌ not solved")
        browser.close()
        return 0 if result.is_solved else 1


if __name__ == "__main__":
    raise SystemExit(main())

File: js/examples/withPlaywright.ts
import { chromium } from 'playwright';
import { CaptchaKrakenSolver } from 'captchakraken';

async function main() {
  const url = process.argv[2] ?? 'https://www.google.com/recaptcha/api2/demo';

  const browser = await chromium.launch({ headless: true });
  const page = await browser.newPage();
  await page.goto(url);

  const solver = new CaptchaKrakenSolver();
  const result = await solver.solve(page);

  console.log(result?.isSolved ? '✅ solved' : '❌ not solved');
  await browser.close();
}

main().catch((e) => { console.error(e); process.exit(1); });
